src/components/Settings/SettingsDialog.tsx

```diff
@@ -65,23 +60,6 @@ export const SettingsDialog = ({ open, onOpenChange }: SettingsDialogProps) => {
     formState: { isDirty, errors },
   } = formMethods
 
-  const onSubmit = async (data: AppSettings) => {
```

Comment:
using tanstack query is more ergonomic and allows us to reused cached settings across different components